impl/src/main/java/com/okta/sdk/impl/resource/identity/provider/DefaultFacebookIdentityProviderBuilder.java

```diff
@@ -0,0 +1,92 @@
+import com.okta.sdk.resource.policy.PolicyAccountLink;
+import com.okta.sdk.resource.policy.PolicySubject;
+import com.okta.sdk.resource.policy.PolicySubjectMatchType;
+import com.okta.sdk.resource.policy.PolicyUserNameTemplate;
+
+import java.util.List;
+
+public class DefaultFacebookIdentityProviderBuilder extends DefaultIdentityProviderBuilder<FacebookIdentityProviderBuilder>
+    implements FacebookIdentityProviderBuilder {
+
+    private Boolean isProfileMaster;
+    private String userNameTemplate;
+
+    @Override
+    public FacebookIdentityProviderBuilder setIsProfileMaster(Boolean isProfileMaster) {
+        this.isProfileMaster = isProfileMaster;
+        return this;
+    }
+
+    @Override
+    public FacebookIdentityProviderBuilder setUserNameTemplate(String userNameTemplate) {
+        this.userNameTemplate = userNameTemplate;
+        return this;
+    }
+
+    @Override
+    public IdentityProvider buildAndCreate(Client client) {
+
+        IdentityProvider createdIdp = client.createIdentityProvider(client.instantiate(IdentityProvider.class)
+            .setType(IdentityProvider.TypeEnum.FACEBOOK)
+            .setName(name)
+            .setProtocol(client.instantiate(Protocol.class)
+                .setType(Protocol.TypeEnum.OAUTH2)
+                .setScopes(scopes)
+                .setCredentials(client.instantiate(IdentityProviderCredentials.class)
+                    .setClient(client.instantiate(IdentityProviderCredentialsClient.class)
+                        .setClientId(clientId)
+                        .setClientSecret(clientSecret))))
+            .setPolicy(client.instantiate(IdentityProviderPolicy.class)
+                .setProvisioning(client.instantiate(Provisioning.class)
+                    .setAction(Provisioning.ActionEnum.AUTO)
+                    .setProfileMaster(isProfileMaster)
+                    .setGroups(client.instantiate(ProvisioningGroups.class)
+                        .setAction(ProvisioningGroups.ActionEnum.NONE))
+                    .setConditions(client.instantiate(ProvisioningConditions.class)
+                        .setDeprovisioned(client.instantiate(ProvisioningDeprovisionedCondition.class)
+                            .setAction(ProvisioningDeprovisionedCondition.ActionEnum.NONE))
+                        .setSuspended(client.instantiate(ProvisioningSuspendedCondition.class)
+                            .setAction(ProvisioningSuspendedCondition.ActionEnum.NONE))))
+                .setAccountLink(client.instantiate(PolicyAccountLink.class)
+                    .setFilter(null)
+                    .setAction(PolicyAccountLink.ActionEnum.AUTO))
+                .setSubject(client.instantiate(PolicySubject.class)
+                    .setUserNameTemplate(client.instantiate(PolicyUserNameTemplate.class)
+                        .setTemplate(userNameTemplate))
+                    .setMatchType(policySubjectMatchType))
+                .setMaxClockSkew(maxClockSkew)));
```

Comment:
There are a bunch of fields available in the IdP creation UI that are not available in this builder (or are hard coded here)

Reply:
let me study the UI and compare each of these builders for missing fields.

Reply:
updates done.